public/components/datasources/components/manage/associated_objects/utils/associated_objects_tab_loading.tsx

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { EuiEmptyPrompt, EuiLoadingSpinner } from '@elastic/eui';
+import React from 'react';
+
+interface AssociatedObjectsTabLoadingProps {
+  objectType: string;
+  warningMessage: boolean;
+}
+
+export const AssociatedObjectsTabLoading: React.FC<AssociatedObjectsTabLoadingProps> = (props) => {
+  const { objectType, warningMessage } = props;
+
+  const BodyText = () => (
+    <>
+      <p>Loading {objectType}</p>
+      {warningMessage ? <p>This may take a moment.</p> : <></>}
+    </>
+  );
+
+  return <EuiEmptyPrompt icon={<EuiLoadingSpinner size="xl" />} body={<BodyText />} />;
```

Comment:
Question: is this size the common alignment in ux design?, as I recall we also have loading spinner with other sizes, not sure we should use xl or align with others